pandas/tests/dtypes/test_common.py

```diff
@@ -731,3 +732,11 @@ def test__is_dtype_type_sparse():
     result = np.dtype("int32")
     assert com._is_dtype_type(ser, lambda tipo: tipo == result)
     assert com._is_dtype_type(ser.dtype, lambda tipo: tipo == result)
+
+
+def test__nansafe_nat_to_int():
```

Comment:
rename to 

test_astype_nansafe; can you parameterize this on all ints at the very least. ideally we would have more testing for this (but can be followup)

Reply:
For the other (non-int64) dtypes we end up hitting the `TypeError` here (related to the above comment):
https://github.com/pandas-dev/pandas/blob/b81f433382e5a35399a12e206178efd58ff9d88a/pandas/core/dtypes/cast.py#L707
Would you recommend breaking these out into a separate test?